oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -90,6 +91,7 @@ abstract static class CredentialSource implements java.io.Serializable {
   private final CredentialSource credentialSource;
   private final Collection<String> scopes;
   private final ServiceAccountImpersonationOptions serviceAccountImpersonationOptions;
+  private ExternalAccountMetricsHandler metricsHandler;
```

Comment:
is this handler really specific to external accounts creds or other creds will have similar thing with different config vars later? 

If the latter we might need to move this to GoogleCredentials later.

Reply:
This handler is specific to external account creds (the header value it's building is only attached to the STS token exchange request for external creds) Other credentials will need header values for x-goog-api-client, but wouldn't be creating or using them in the same way.